pkg/storegateway/indexheader/encoding/factory_test.go

```diff
@@ -9,154 +9,185 @@ import (
 	"path"
 	"testing"
 
+	"github.com/prometheus/client_golang/prometheus"
 	promencoding "github.com/prometheus/prometheus/tsdb/encoding"
 	"github.com/stretchr/testify/require"
 )
 
```

Comment:
I would recommend to add a unit test on concurrency too. For reference, something like the one I've recently done here: https://github.com/grafana/mimir/pull/3666/files#diff-277ce76efcdd7624ec957cbb2be9869d235828dd2252cda8c9e812a14a97ded1R590

Reply:
Done in c9096e9aec8b258982218661b9195456e3b679f0